lib/view.dart

```diff
@@ -156,49 +156,109 @@ void renderChartOptions(bool isComparisonChecked, bool isNormalisationChecked) {
+      'comparison-option', 'Compare data', isComparisonChecked,
+      (bool isChecked) {
+    command(UIAction.toggleDataComparison, ToggleOptionEnabledData(isChecked));
+  });
+  optionsCol.append(comparisonCheckbox);
+
+  var normalisationCheckbox = _getCheckboxWithLabel(
+      'normalisation-option', 'Normalise data', isNormalisationChecked,
+      (bool isChecked) {
+    command(
+        UIAction.toggleDataNormalisation, ToggleOptionEnabledData(isChecked));
+  });
+  optionsCol.append(normalisationCheckbox);
+
+  wrapper.append(labelCol);
+  wrapper.append(optionsCol);
+  content.append(wrapper);
+}
+
+void renderFilterDropdowns(
+    List<String> filterKeys,
+    Map<String, List<String>> filterOptions,
+    bool shouldRenderComparisonFilters) {
+  var wrapper = getGridRow();
+  var labelCol = getGridLabelCol()..innerText = 'Filters';
+  var optionsCol = getGridOptionsCol();
+
+  for (var key in filterKeys) {
+    var filterRow = getGridRow();
+    var checkboxCol = html.DivElement()..classes = ['col-3'];
+    var filterCol = html.DivElement()..classes = ['col-3'];
+
+    var checkboxWithLabel = _getCheckboxWithLabel(
+        'filter-option-${key}', key, false, (bool isChecked) {
+      command(
+          UIAction.toggleActiveFilter, ToggleActiveFilterData(key, isChecked));
     });
-  var comparisonLabel = html.LabelElement()
-    ..htmlFor = 'data-comparison-option'
-    ..classes = ['form-check-label']
-    ..innerText = 'Compare data';
-  comparisonWrapper.append(comparisonOption);
-  comparisonWrapper.append(comparisonLabel);
-  optionsCol.append(comparisonWrapper);
+    checkboxCol.append(checkboxWithLabel);
 
-  var normalisationWrapper = html.DivElement()
-    ..classes = ['form-check', 'form-check-inline'];
-  var normalisationOption = html.InputElement()
-    ..type = 'checkbox'
-    ..id = 'data-normalisation-option'
-    ..classes = ['form-check-input']
-    ..checked = isNormalisationChecked
-    ..onChange.listen((e) {
-      command(UIAction.toggleDataNormalisation,
-          ToggleOptionEnabledData((e.target as html.CheckboxInputElement).checked));
+    var filterDropdown =
+        _getDropdown(filterOptions[key].toList(), '__all', (String value) {
+      command(UIAction.setFilterValue, SetFilterValueData(key, value));
     });
-  var normalisationLabel = html.LabelElement()
-    ..htmlFor = 'data-normalisation-option'
-    ..classes = ['form-check-label']
-    ..innerText = 'Normalise data';
-  normalisationWrapper.append(normalisationOption);
-  normalisationWrapper.append(normalisationLabel);
-  optionsCol.append(normalisationWrapper);
+    filterCol.append(filterDropdown);
+
+    filterRow.append(checkboxCol);
+    filterRow.append(filterCol);
+    optionsCol.append(filterRow);
+
+    if (!shouldRenderComparisonFilters) continue;
+    var comparisonFilterCol = html.DivElement()..classes = ['col-3'];
+    var comparisonFilterDropdown =
+        _getDropdown(filterOptions[key].toList(), '__all', (String value) {
```

Comment:
If I understand correctly, `__all` is the default value - which is added in the controller to all of the filter option lists, and also passed here as the value to be selected. Would it be worth defining it only in one place (ie. the controller) and passing it down to `renderFilterDropdowns` as an argument?

Reply:
So adding '__all' to the values passed to the view - I'm adding it in the controller. This __all is to say the default value that needs to be selected when rendering. This parameter in fact should be passed from the controller based on the selected filter. So would it be okay if i fix it when i bring in that functionality?